server/src/main/java/org/elasticsearch/action/search/CountResponse.java

```diff
@@ -0,0 +1,232 @@
+        public static final ParseField TOTAL = new ParseField("total");
+        public static final ParseField SUCCESSFUL = new ParseField("successful");
+        public static final ParseField FAILURES = new ParseField("failures");
+
+        private final int successfulShards;
+        private final int totalShards;
+        private final int skippedShards;
+        private ShardSearchFailure[] shardFailures;
+
+        public ShardStats(int successfulShards, int totalShards, int skippedShards, ShardSearchFailure[] shardFailures) {
+            this.successfulShards = successfulShards;
+            this.totalShards = totalShards;
+            this.skippedShards = skippedShards;
+            this.shardFailures = shardFailures;
+        }
+
+        public static ShardStats fromXContent(XContentParser parser) throws IOException {
+            int successfulShards = -1;
+            int totalShards = -1;
+            int skippedShards = 0; //BWC @see org.elasticsearch.action.search.SearchResponse
+            List<ShardSearchFailure> failures = new ArrayList<>();
+            XContentParser.Token token;
+            String currentName = parser.currentName();
+            while ((token = parser.nextToken()) != XContentParser.Token.END_OBJECT) {
+                if (token == XContentParser.Token.FIELD_NAME) {
+                    currentName = parser.currentName();
+                } else if (token.isValue()) {
+                    if (FAILED.match(currentName, parser.getDeprecationHandler())) {
+                        parser.intValue();
+                    } else if (SKIPPED.match(currentName, parser.getDeprecationHandler())) {
+                        skippedShards = parser.intValue();
+                    } else if (TOTAL.match(currentName, parser.getDeprecationHandler())) {
+                        totalShards = parser.intValue();
+                    } else if (SUCCESSFUL.match(currentName, parser.getDeprecationHandler())) {
+                        successfulShards = parser.intValue();
+                    } else {
+                        parser.skipChildren();
+                    }
+                } else if (token == XContentParser.Token.START_ARRAY) {
+                    if (FAILURES.match(currentName, parser.getDeprecationHandler())) {
+                        while ((parser.nextToken()) != XContentParser.Token.END_ARRAY) {
+                            failures.add(ShardSearchFailure.fromXContent(parser));
+                        }
+                    } else {
+                        parser.skipChildren();
+                    }
+                } else {
+                    parser.skipChildren();
+                }
+            }
+            return new ShardStats(successfulShards,totalShards,skippedShards,failures.toArray(new ShardSearchFailure[failures.size()]));
+        }
+
+        @Override
+        public XContentBuilder toXContent(XContentBuilder builder, Params params) throws IOException {
```

Comment:
the XContent here does not match what you removed in the REST API. There was a bit about early termination, as well as count that you need to include. You likely need to also include the begin/end calls, or else this will fail tests.

You can check the tests with `./gradlew :server:check` from the base of the checkout. If tests dont fail then we need some better tests around the response hehe

Reply:
Hey think you're looking at `XContent` of `ShardStats` (which innner class of `CountRequest`), and `ShardStats` indeed handles only the `_shards` part of the response.

`count` is added up at `line 150` of `CountRequest`, `terminated_early` at `line 152`, and then at `line 154` `CountRequest` will make a call to inner class `ShardStats.toXContent`

I've added the radnomized tests which also include `ShardFailures` and `terminated_early`, and `./gradlew clean server:check` runs fine

So as much as I'm aware of this part is ok (at least regarding this comment), can you please check again?

Reply:
Hi, I was referring to this code, https://github.com/elastic/elasticsearch/blob/master/server/src/main/java/org/elasticsearch/rest/action/document/RestCountAction.java#L95-L104 , which you deleted as part of your PR.

Reply:
👍